lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -47,15 +47,25 @@ def capture(money, transaction_id, options={})
         end
       end
 
-      def purchase(money, card_or_token, options={})
-        commit('CreditSale') do |xml|
-          add_amount(xml, money)
-          add_allow_dup(xml)
-          add_customer_data(xml, card_or_token, options)
-          add_details(xml, options)
-          add_descriptor_name(xml, options)
-          add_payment(xml, card_or_token, options)
-          add_three_d_secure(xml, card_or_token, options)
+      def purchase(money, payment_method, options={})
+        if payment_method.is_a?(Check)
```

Comment:
Just a nitpicky style idea, but might be nice to break these two different paths up into two methods (maybe like `def purchase_with_check` and `def purchase_with_credit`). It'd shorten this method up a bit. What do you think?

Reply:
Seems reasonable; now done :)